docs/decisions/backend/0010-advanced-installer-suggestions.md

```diff
@@ -0,0 +1,123 @@
+
+    /// <summary>
+    /// List of known recognizable file extensions for direct children of the target <see cref="DestinationGamePath"/>.
+    /// NOTE: Only include file extensions that are only likely to appear at this level of the folder hierarchy.
+    /// </summary>
+    public IEnumerable<Extension> KnownValidFileExtensions { get; init; }
+
+    /// <summary>
+    /// List of file extensions to discard when installing to this target.
+    /// </summary>
+    public IEnumerable<Extension> FileExtensionsToDiscard { get; init; }
+
+    /// <summary>
+    /// Collection of Targets that are nested paths relative to <see cref="DestinationGamePath"/>.
+    /// </summary>
+    public IEnumerable<ISuggestionFolder> SubTargets { get; init; }
+}
+```
+
+## Acquiring `ISuggestionFolder`(s) During Deploy Step (a.k.a. `GetModsAsync`)
+
+Extend the `IGame` interface to expose a property which returns `ISuggestionFolder`(s) for the game's most common directories.
+
+This property is populated with the following elements:
+- All `GamePath` entries (i.e. Game folder, Save folder, Config folder, etc.)
+- All `InstallFolderTarget` entries (e.g. `Data` folder for Skyrim.), as `ISuggestionFolder`.
+- Custom `ISuggestionFolder`(s) defined on a per game basis.
+- Apply the filtering steps detailed below. (Not MVP, will be iteratively added later)
+
+This interface can be accessed during the deploy step under `GameInstallation` structure.
+
+## Filtering Suggestions (Directories)
+
+Note: String comparisons performed here adhere to ![Paths Doc](./0003-paths.md). [Case insensitive + / separator]
+
+The suggestion algorithm for folders is described below.
+We will use modified example from [Advanced Installer Design](./0009-advanced-installer-design.md).
+
+Consider the following directory structure:
+
+```text
+Red Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Green Hair
+├─ Data
+│  ├─ model.dae
+│  └─ hair.dds
+Blue Hair
+└─ Data
+   ├─ model.dae
+   └─ hair.dds
+```
+
+Assume the `Data` folder is a folder within the game directory (also applies to subfolders).
+
+If the user selects `model.dae` in `Blue Hair`; the suggestion algorithm should try finding the following in the game's directory tree:
+- `Blue Hair/Data`
+- `Data`
+
+We keep removing the first directory in the path, until there's at least 1 match in the game's directory tree
+(substring sanitized path!). Once there is at least 1 match, include that in the suggestions and terminate the search.
+
+tl;dr Include any folders where any `substring` of the in-archive directory path, matches an `InstallFolderTarget`.
+
+## Filtering Suggestions (Files)
+
+The filtering algorithm for suggestions where to place a file functions in the following manner:
+
+- Filter by `KnownValidFileExtensions`. (Remove if unknown extension).
+- Filter by `FileExtensionsToDiscard`. (Remove if extension to discard).
+- Add any directories from [Implementation Algorithm (Directories)](#implementation-algorithm-directories)
```

Comment:
Remember that the whole point of the manual installer is to allow users to install stuff wherever, we can't filter out options for them. 

Honestly the best thing we could possibly ever do in the future is to sort the options in order by most likely, potentially show that as a special suggested option at the top. 
But all of this should first go through Design, we should not include this ADR as if we already accepted this idea.
I think you should just keep the parts about how to get the target locations, remove everything else as this is not something we discussed or agreed on. 

If you want you can open an issue to request sorting of the target locations, so that design can take a look at it.

Reply:
Note: We've decided to put the non-core ADRs as separate issues; so they can be merged with their corresponding implementations later. The ADR text will be preserved inside each issue.